File: app/src/main/java/io/inabsentia/superhangman/activity/MatchHistoryActivity.java
package io.inabsentia.superhangman.activity;

import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.List;

import io.inabsentia.superhangman.R;
import io.inabsentia.superhangman.data.dao.IMatchDAO;
import io.inabsentia.superhangman.data.dao.MatchDAO;
import io.inabsentia.superhangman.data.dto.MatchDTO;

public class MatchHistoryActivity extends AppCompatActivity implements View.OnClickListener {

    private TextView tvCustomTitle;
    private Button btnReset;

    private ListView matchListView;
    private ArrayAdapter listViewAdapter;

    private List<MatchDTO> matchList;

    private IMatchDAO matchDAO = MatchDAO.getInstance();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.match_history_activity);
        getSupportActionBar().setDisplayOptions(ActionBar.DISPLAY_SHOW_CUSTOM);
        getSupportActionBar().setCustomView(R.layout.custom_action_bar);

        /*
         * Get all the high scores.
         */
        try {
            matchList = matchDAO.getAll();
        } catch (IMatchDAO.DALException e) {
            e.printStackTrace();
        }

        /*
         * Instantiate objects.
         */
        tvCustomTitle = (TextView) findViewById(R.id.action_bar_title);
        btnReset = (Button) findViewById(R.id.btn_reset_match_history);
        matchListView = (ListView) findViewById(R.id.high_score_list);

        listViewAdapter = new ArrayAdapter(this, android.R.layout.simple_list_item_1, matchList);
        matchListView.setAdapter(listViewAdapter);

        /* Set title of action bar */
        tvCustomTitle.setText(R.string.match_history_title_label);

        /*
         * Set I/O listeners.
         */
        btnReset.setOnClickListener(this);

        /*
         * Update the display.
         */
        updateDisplay();
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btn_reset_match_history:
                resetHistory();
                updateDisplay();
                break;
            default:
                break;
        }
    }

    private void updateDisplay() {
        try {
            matchList = matchDAO.getAll();
            matchListView.removeAllViewsInLayout();
        } catch (IMatchDAO.DALException e) {
            e.printStackTrace();
        }
    }

    private void resetHistory() {
        try {
            matchDAO.removeAll();
            matchDAO.save(getApplicationContext());
        } catch (IMatchDAO.DALException e) {
            Toast.makeText(getApplicationContext(), R.string.match_history_reset_failed, Toast.LENGTH_LONG).show();
            e.printStackTrace();
        }
    }

}

File: app/src/main/java/io/inabsentia/superhangman/activity/MatchHistoryActivity.kt
package io.inabsentia.superhangman.activity

import android.os.Bundle
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.ListView
import android.widget.TextView
import io.inabsentia.superhangman.R
import io.inabsentia.superhangman.adapter.MatchHistoryAdapter
import io.inabsentia.superhangman.data.dao.MatchDAO
import io.inabsentia.superhangman.data.dto.MatchDTO

class MatchHistoryActivity : AppCompatActivity() {

    private var tvCustomTitle: TextView? = null

    private var matchListView: ListView? = null
    private var matchAdapter: MatchHistoryAdapter? = null

    private var matchList: List<MatchDTO>? = null

    private val matchDAO = MatchDAO.instance

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.match_history_activity)
        supportActionBar!!.displayOptions = ActionBar.DISPLAY_SHOW_CUSTOM
        supportActionBar!!.setCustomView(R.layout.custom_action_bar)

        /*
         * Get all the high scores.
         */
        matchList = matchDAO!!.all

        /*
         * Instantiate objects.
         */
        tvCustomTitle = findViewById(R.id.action_bar_title)
        matchListView = findViewById(R.id.match_history_list)
        val emptyText: TextView = findViewById(R.id.empty_text)

        /* Remove placeholder text if list is not empty */
        if (matchList!!.isNotEmpty())
            emptyText.visibility = View.INVISIBLE

        matchAdapter = MatchHistoryAdapter(matchList!!, this)
        matchListView!!.adapter = matchAdapter

        /* Set title of action bar */
        tvCustomTitle!!.setText(R.string.match_history_title_label)

        /*
         * Update the display.
         */
        updateDisplay()
    }

    private fun updateDisplay() {
        matchList = matchDAO!!.all
        matchListView!!.removeAllViewsInLayout()
    }

}